x-pack/plugins/lens/public/heatmap_visualization/suggestions.ts

```diff
@@ -58,9 +58,17 @@ export const getSuggestions: Visualization<HeatmapVisualizationState>['getSugges
    * Hide for:
    * - reduced and reorder tables
    * - tables with just a single bucket dimension
+   * - tables with a date histogram
    */
+  const hasOnlyOneDatehistogramBucket =
+    groups.some((group) => group.operation.dataType === 'date') &&
+    metrics.length === 1 &&
+    groups.length === 1;
```

Comment:
let's discuss this condition during sync, but if we decide to stick to this solution, I think it'll be better to change the order of execution and check if all groups are date_histogram:

```suggestion
  const hasOnlyDatehistogramBuckets =
    metrics.length === 1 &&
    groups.length > 0 &&
    groups.every((group) => group.operation.dataType === 'date')
```